### ISSUE TITLE
Google Drive hotlinking support

### ISSUE THREAD

**Issue by Copytalky** [dularion/streama#60]
I've yet to find a media server app that allows the offloading of transcoding to be handled by Google Drive via hotlinking media.

I currently use Google Drive for my storage needs of my media due to travelling around a lot so I find it cumbersome to carry around extra drives.

Google Drive does all the transcoding work for me, depending where I stream from, I can select what quality I want to receive. This puts no pressure on my system and I don't have to keep a server running at home and pay for a beefed up internet connection to stream abroad. Google handles all of that for me.

What I can't do with Google Drive is have a nice UI like streama.

If Streama could put a feature in where you can host a streama VPS and hotlink to Google Drive accounts, I think that would be a really great feature to have. For people that want to use streama outside the home and don't have the internet connection bandwidth to stream abroad in high quality, Google Drive is one of the best options. (clouds in general). I'd be willing to pay for a cheap VPS to have streama running on where I could login to it anywhere I am and stream my hotlinked Google Drive files.

Also I have a couple of Google Drive accounts so it would be great to have support for multi-account importing and even just an input field where you can input a google drive hotlinked URL would be sufficient.

**Comment by admwx7** [comment]
This has been touched on in #15 to an extent.  Though this would be a better solution than generic URL based third party linking.  However, at that point you're essentially just wrapping their API with a pretty skin, don't know Google's thoughts on that, might want to check into their licensing first.  It would play nicely with chromecast though I bet.

**Comment by Copytalky** [comment]
From my understanding it would be 'ok' with Google. This is a open-sourced platform afterall. Not like a paid platform that's ripping google off by utilizing their google drive services. There are lots of apps out there as well which utilize google drive for backing up config data, syncing videos/photos/files etc, and those apps are sometimes paid ones.

Other platforms like Emby utilize google drive for storing transcoded video files, the problem with that is you have to have an Emby server, store the video file, have Emby transcode it, have Emby then upload the transcodes to Google Drive. When I'm on the go I don't want to have a media server at home transcoding a media file then uploading to google drive, I see it as nailing two birds with one stone if you just upload your media to Google Drive and utilize their player for handling transcoding.

**Comment by admwx7** [comment]
It sounds to me more like you just want the UI from streama connected to 1+ google accounts instead of having a backend that can stream to some device, whether it's the one that's viewing the UI or a chromecast or something along those lines.  At that point it would make more sense to create a LITE mobile version that would allow you to connect drive accounts that are stored on the device, then connects to drive and stores meta data about watched status back to drive, and would use an html5 player or connect to a chromecast to stream directly from drive.  Would remove the need for a VPS since everything would be run locally on your phone or streaming through drive.

I haven't looked at Google's licensing for drive so I'm not going to claim knowledge of their T&C for usage and streaming from their platform, but my guess would be that the responsibility all falls on the user, not the developer as long as you're not trying to do anything malicious, just felt the need to mention that it's something to keep in mind when you're looking to interface with another service like that.

**Comment by Copytalky** [comment]
yeah you are right in the sense that you don't really need a VPS then. rather just something like a cordova based app platform that has a nice UI and allows you to connect to your google drive accounts and import the hotlinks.

**Comment by admwx7** [comment]
Would be pretty easy to have the UI save to local storage and the api at
the same time and fetch from both, api getting priority if it's configured,
then running mobile or desktop without a configured server back-end would
provide that same functionality without a lot of extra code.

**Comment by pmjm** [comment]
Wondering if this feature was still being considered, as it would benefit me as well. It would be amazing just to be able to hotlink the streamed videos (in varying qualities) from Google Drive directly to the HTML5 player, saving the bandwidth of the strema server.

**Comment by 4Sol** [comment]
Updates?
